In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
import pandas as pd

### 행정동별 독거노인 수와 병원 수

#### 행정동별 병원 수

In [29]:
csv_path_medical = '/content/drive/MyDrive/town_cc_big_data/data/행정동_강원도_춘천시_의료기관약국_20220805.csv'
df_medical_raw = pd.read_csv(csv_path_medical)

address_data = df_medical_raw['행정동']
kind_data = df_medical_raw['종류']

df_medical = pd.DataFrame({'종류': kind_data, '행정동명': address_data})

count_by_dong = df_medical.groupby('행정동명').count().rename(columns={'종류': '병원 수'})
count_by_dong = count_by_dong.sort_values('병원 수', ascending=True)
count_by_dong = count_by_dong.head(18)

#### 행정동별 독거노인 수

In [30]:
csv_path_old = '/content/drive/MyDrive/town_cc_big_data/data/강원도 춘천시_독거노인현황_20220620.csv'
df_old_raw = pd.read_csv(csv_path_old, encoding='euc-kr')

df_old = df_old_raw.groupby('행정동명')['인구수'].sum().reset_index()
df_old = df_old.sort_values('인구수', ascending = False)

df_old = df_old.head(10)

#### 두 테이블 병합

In [33]:
df_old

행정동명   인구수
18   퇴계동  1538
11   석사동  1316
24  후평3동  1168
15  신사우동  1044
20  효자2동  1019
22  후평1동   848
0    강남동   799
6     동면   769
5    동내면   677
14   신북읍   643

In [32]:
count_by_dong

병원 수
행정동명      
신동       1
약사동      1
요선동      1
삼천동      1
봉의동      1
동산면      1
동산명      1
남면       1
호반동      1
근화동      1
중앙동      1
칠전동      2
남산면      2
서면       2
신동면      2
신북읍      3
사농동      3
약사명동     3

In [22]:
medical_old = pd.merge(df_old, count_by_dong, how='inner', left_on = '행정동명', right_on = '행정동명')
medical_old

행정동명  인구수  병원 수
0  신북읍  643     3

신북읍에 1인 독거노인 인구수 643명이 사는데 병원 수는 3개 밖에 없음.

---